components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -32,13 +50,230 @@ func (ss *BatchCheckService) BatchCheck(checks []string, config models.Config) m
 		checkTriggerRespMap = getBastionCheckResp(ss, bastionChecks, config)
 	}
 
-	// Get remote check trigger resp and append to checkTriggerRespMap (keys and values)
+	successfullyStartedMockServers, notStartedMockServers, err := ss.startMockServerIfNeeded(remoteChecks, config)
+	if err != nil {
+		ss.log.Error("Error occurred while starting mock server")
+		return models.BatchCheckResponse{}, err
+	}
+
+	if len(notStartedMockServers) > 0 {
+		for _, successfullyStartedMockServer := range successfullyStartedMockServers {
+			if successfullyStartedMockServer.Host != "" {
+				go ss.StopMockServerOnHostAndPort(successfullyStartedMockServer.Host, successfullyStartedMockServer.Protocol, successfullyStartedMockServer.Port)
```

Comment:
Why do we need to make this as a Goroutine ? After this we are exiting only, why cant we wait ?

Reply:
There are multiple mock server running on a node. so making api calls in parallel. Also now I am waiting for response from go routine so this will wait first